From b63192ed67cfb33244ebd91b23423d7c98af1d55 Mon Sep 17 00:00:00 2001
From: Zbigniew Kurzynski <zbigniew.kurzynski@intel.com>
Date: Tue, 15 Nov 2022 14:06:14 +0100
Subject: [PATCH] Using local power.hpp file instead of upstreamed

This patch is replacing usage of file the file redfish-core/lib/power.hpp
with a local file redfish-core/lib/node-manager/power.hpp

This patch also is enabling depracted Power schema.
Upstream-Status: Pending
---
 redfish-core/include/redfish.hpp | 4 ++--
 redfish-core/lib/chassis.hpp     | 2 +-
 2 files changed, 3 insertions(+), 3 deletions(-)

diff --git a/redfish-core/include/redfish.hpp b/redfish-core/include/redfish.hpp
index 2cd1357f..f8a6ba2d 100644
--- a/redfish-core/include/redfish.hpp
+++ b/redfish-core/include/redfish.hpp
@@ -40,9 +40,9 @@
 #include "metric_report.hpp"
 #include "metric_report_definition.hpp"
 #include "network_protocol.hpp"
+#include "node-manager/power.hpp"
 #include "pcie.hpp"
 #include "pcie_slots.hpp"
-#include "power.hpp"
 #include "power_subsystem.hpp"
 #include "power_supply.hpp"
 #include "processor.hpp"
@@ -93,8 +93,8 @@ class RedfishService
         requestEthernetInterfacesRoutes(app);
 #ifdef BMCWEB_ALLOW_DEPRECATED_POWER_THERMAL
         requestRoutesThermal(app);
-        requestRoutesPower(app);
 #endif
+        requestRoutesPower(app);
 #ifdef BMCWEB_NEW_POWERSUBSYSTEM_THERMALSUBSYSTEM
         requestRoutesEnvironmentMetrics(app);
         requestRoutesPowerSubsystem(app);
diff --git a/redfish-core/lib/chassis.hpp b/redfish-core/lib/chassis.hpp
index a2744210..6ff2e4bb 100644
--- a/redfish-core/lib/chassis.hpp
+++ b/redfish-core/lib/chassis.hpp
@@ -425,11 +425,11 @@ inline void
                 asyncResp->res.jsonValue["Thermal"]["@odata.id"] =
                     boost::urls::format("/redfish/v1/Chassis/{}/Thermal",
                                         chassisId);
+#endif
                 // Power object
                 asyncResp->res.jsonValue["Power"]["@odata.id"] =
                     boost::urls::format("/redfish/v1/Chassis/{}/Power",
                                         chassisId);
-#endif
 #ifdef BMCWEB_NEW_POWERSUBSYSTEM_THERMALSUBSYSTEM
                 asyncResp->res.jsonValue["ThermalSubsystem"]["@odata.id"] =
                     boost::urls::format(
-- 
2.25.1

